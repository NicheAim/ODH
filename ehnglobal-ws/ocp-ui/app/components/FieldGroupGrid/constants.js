export const PREFIX = 'fieldGroupPrefix';
export const MAIN = 'fieldGroupMain';
export const SUFFIX = 'fieldGroupSuffix';
