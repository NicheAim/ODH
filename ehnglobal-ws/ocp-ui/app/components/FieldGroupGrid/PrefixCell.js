import React from 'react';
import PropTypes from 'prop-types';
import { Cell } from 'styled-css-grid';

import { PREFIX } from './constants';

function PrefixCell({ area, children, ...rest }) {
  return (
    <Cell area={area} {...rest}>
      {children}
    </Cell>);
}

PrefixCell.propTypes = {
  ...Cell.propTypes,
  children: PropTypes.node,
  area: PropTypes.string,
};

PrefixCell.defaultProps = {
  area: PREFIX,
};

export default PrefixCell;
