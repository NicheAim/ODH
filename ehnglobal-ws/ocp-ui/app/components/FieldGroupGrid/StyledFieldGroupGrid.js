import PropTypes from 'prop-types';
import { Grid } from 'styled-css-grid';
import { PREFIX, MAIN, SUFFIX } from './constants';

const StyledFieldGroupGrid = Grid.extend`
  grid-template-columns: 1fr;
  grid-template-areas: "${PREFIX}" "${MAIN}"${(props) => props.withSuffix ? ` "${SUFFIX}"` : ''};

  @media (min-width: 768px) {
    grid-template-columns: ${(props) => props.withSuffix ? '1fr 2fr 1fr' : '1fr 2fr'};
    grid-template-areas: "${PREFIX} ${MAIN}${({ withSuffix }) => withSuffix ? ` ${SUFFIX}` : ''}";
  }
`;

StyledFieldGroupGrid.propTypes = {
  ...Grid.propTypes,
  withSuffix: PropTypes.bool,
};

StyledFieldGroupGrid.defaultProps = {
  withSuffix: false,
};

export default StyledFieldGroupGrid;
