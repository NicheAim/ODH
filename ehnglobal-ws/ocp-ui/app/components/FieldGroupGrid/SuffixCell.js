import React from 'react';
import PropTypes from 'prop-types';
import { Cell } from 'styled-css-grid';

import { SUFFIX } from './constants';

function SuffixCell({ area, children, ...rest }) {
  return (
    <Cell area={area} {...rest}>
      {children}
    </Cell>);
}

SuffixCell.propTypes = {
  ...Cell.propTypes,
  children: PropTypes.node,
  area: PropTypes.string,
};

SuffixCell.defaultProps = {
  area: SUFFIX,
};

export default SuffixCell;
