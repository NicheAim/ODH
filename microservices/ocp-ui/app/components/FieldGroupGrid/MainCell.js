import React from 'react';
import PropTypes from 'prop-types';
import { Cell } from 'styled-css-grid';

import { MAIN } from './constants';

function MainCell({ area, children, ...rest }) {
  return (
    <Cell area={area} {...rest}>
      {children}
    </Cell>);
}

MainCell.propTypes = {
  ...Cell.propTypes,
  children: PropTypes.node,
  area: PropTypes.string,
};

MainCell.defaultProps = {
  area: MAIN,
};

export default MainCell;
