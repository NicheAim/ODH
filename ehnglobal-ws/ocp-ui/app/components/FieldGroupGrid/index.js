/**
 *
 * IdentifierGroupGrid
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import StyledFieldGroupGrid from './StyledFieldGroupGrid';

function IdentifierGroupGrid({ children, gap, withSuffix, ...rest }) {
  return (
    <StyledFieldGroupGrid gap={gap} withSuffix={withSuffix} {...rest}>
      {children}
    </StyledFieldGroupGrid>
  );
}

IdentifierGroupGrid.propTypes = {
  ...StyledFieldGroupGrid.propTypes,
  children: PropTypes.node,
  gap: PropTypes.string,
  withSuffix: PropTypes.bool,
};

IdentifierGroupGrid.defaultProps = {
  gap: '0',
  withSuffix: false,
};

export default IdentifierGroupGrid;
