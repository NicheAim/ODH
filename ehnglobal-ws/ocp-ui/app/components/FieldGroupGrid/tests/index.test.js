import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';

import FieldGroupGrid from '../index';

configure({ adapter: new Adapter() });

describe('<FieldGroupGrid />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot with no gap prop', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<FieldGroupGrid>{children}</FieldGroupGrid>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });

    it('should match snapshot with gap prop', () => {
      // Arrange
      const children = <div>test</div>;
      const gap = '10';

      // Act
      const renderedComponent = shallow(<FieldGroupGrid gap={gap}>{children}</FieldGroupGrid>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should have children', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<FieldGroupGrid>{children}</FieldGroupGrid>);

      // Assert
      expect(renderedComponent.contains(children)).toEqual(true);
    });

    it('should default gap to 0', () => {
      // Arrange
      const children = <div>test</div>;
      const defaultGap = '0';

      // Act
      const renderedComponent = shallow(<FieldGroupGrid>{children}</FieldGroupGrid>);

      // Assert
      expect(renderedComponent.props().gap).toEqual(defaultGap);
    });

    it('should use pass the configured gap', () => {
      // Arrange
      const children = <div>test</div>;
      const gap = '10';

      // Act
      const renderedComponent = shallow(<FieldGroupGrid gap={gap}>{children}</FieldGroupGrid>);

      // Assert
      expect(renderedComponent.props().gap).toEqual(gap);
    });
  });

  describe('style tests', () => {
    it('should be grid', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<FieldGroupGrid>{children}</FieldGroupGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('display', 'grid');
    });
  });
});
