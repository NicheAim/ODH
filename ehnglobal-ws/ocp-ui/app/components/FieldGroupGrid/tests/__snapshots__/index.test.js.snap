// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldGroupGrid /> snapshot tests should match snapshot with gap prop 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdentifierGroupGrid
    gap="10"
    withSuffix={false}
  >
    <div>
      test
    </div>
  </IdentifierGroupGrid>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        test
      </div>,
      "gap": "10",
      "withSuffix": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          test
        </div>,
        "gap": "10",
        "withSuffix": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<FieldGroupGrid /> snapshot tests should match snapshot with no gap prop 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdentifierGroupGrid
    gap="0"
    withSuffix={false}
  >
    <div>
      test
    </div>
  </IdentifierGroupGrid>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        test
      </div>,
      "gap": "0",
      "withSuffix": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          test
        </div>,
        "gap": "0",
        "withSuffix": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
